File: src/Producto.java
public class Producto {
    private static int CANTIDAD_PRODUCTOS = 0;
    private int numeroSerie;
    private String nombre;
    private double precio;
    private String fechaImportacion;
    private int stock;
    public double getPrecio() {
        return precio;
    }
    public int getStock() {
        return stock;
    }
    public void comprarProducto(int cant) {
        stock-=cant;
    }
    public Producto(String nombre, double precio, String fechaImportacion, int stock) {
        this.numeroSerie = CANTIDAD_PRODUCTOS++;
        this.nombre = nombre;
        this.precio = precio;
        this.fechaImportacion = fechaImportacion;
        this.stock = stock;
        CANTIDAD_PRODUCTOS++;
    }
    protected String obtenerInformacion() {
        return String.format("Nombre: %s, precio %f, fecha de importación: %s, stock: %d",
                nombre, precio, fechaImportacion, stock);
    }
    public void AgregarStock(int cantidad){
        stock+=cantidad;
    }
    public void quitarDeStock(int cantidad){
        stock-=cantidad;
    }
}

File: src/Producto.py

class Producto:
    CANTIDAD_PRODUCTOS = 0

    def __init__(self, nombre, precio, fecha_importacion, stock):
        self.numero_serie = Producto.CANTIDAD_PRODUCTOS
        Producto.CANTIDAD_PRODUCTOS += 1
        self.nombre = nombre
        self.precio = precio
        self.fecha_importacion = fecha_importacion
        self.stock = stock

    def get_precio(self):
        return self.precio

    def get_stock(self):
        return self.stock

    def comprar_producto(self, cant):
        self.stock -= cant

    def obtener_informacion(self):
        return f"Nombre: {self.nombre}, precio: {self.precio}, fecha de importación: {self.fecha_importacion}, stock: {self.stock}"

    def agregar_stock(self, cantidad):
        self.stock += cantidad

    def quitar_de_stock(self, cantidad):
        self.stock -= cantidad